File: AntibodyPropertiesScripts/mutation_frequency69017.r
#Using shazam to generate mutation frequency
#slancast@scg4.stanford.edu:/srv/gsfs0/projects/snyder/slancast/69-017/mutation_freq
#

library(alakazam)
library(shazam)
library(ggplot2)

Subject <- Sys.getenv(c("var"))
output <- substr(Subject, 6, 9)

print("T1")
db <- readChangeoDb(Subject)
ID <- c()
Isotype <- c()
for (i in 1:nrow(db))  { ID <- append(ID,Subject) }
db <- cbind(db, ID)

db_new <- calcDBObservedMutations(db, sequenceColumn="SEQUENCE_IMGT", germlineColumn="GERMLINE_IMGT_D_MASK", frequency=FALSE,regionDefinition=IMGT_V_BY_REGIONS)
print("mutations_calculated")
ls
print("db type")
print(typeof(db_new))
#CDR3_Mut <- matrix(nrow = 0,ncol = 63)
#colnames(CDR3_Mut) <- c(colnames(db_new), "Mutations") #I don't think I use this anymore.

print(ncol(db_new))

write.table(db_new, file=paste(output,"db_new.csv", sep=""), sep=",")

File: AntibodyPropertiesScripts/mutation_frequency69017.py
#slancast@scg4.stanford.edu:/srv/gsfs0/projects/snyder/slancast/69-017/mutation_freq

import pandas as pd
import os

Subject = str(os.environ['var'][5:9])

df = pd.read_csv(Subject+"db_new.csv")

df = pd.DataFrame(df)

df["Total_Mutations"] = df["OBSERVED_FWR1_R"] + df["OBSERVED_CDR1_R"] + df["OBSERVED_FWR2_R"] + df["OBSERVED_CDR2_R"] + df["OBSERVED_FWR3_R"] + df["OBSERVED_CDR3_R"]

df.to_csv(str(os.environ['var'])+"CDR3_Mut.csv")